### PR TITLE
[Feature] adjust iceberg table sink dop

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/21502

In the first milestone of iceberg table sink, we want to support users manually config the sink dop. it mainly to used to increase concurrency and reducing small files by user. If user doesn't configure, we will use the current sink dop default value.  the default value is `min(32, default_dop_by_cores)`.  32 is an empirical value for accessing S3, which is a turning point for the IO concurrency. 
Session variable `pipeline_sink_dop` has been deprecated currently. In this patch, we cancel the deprecation. It only be used in iceberg table sink. We didn't modify any logic of the olap table sink.

We will make adaptive adjustments to the Iceberg sink DOP in the next milestone.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.nio.file.NoSuchFileException: /home/runner/_work/starrocks/starrocks/fe/fe-core/target/result

File: fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java [hunk 1]
```diff
@@ -1725,20 +1740,31 @@ TExecBatchPlanFragmentsParams toThriftInBatch(
...
                 }
-                int curTabletSinkDop = 0;
+                int curTableSinkDop = 0;
                 if (forceSetTableSinkDop) {
-                    curTabletSinkDop = fragment.getPipelineDop();
+                    DataSink dataSink = fragment.getSink();
+                    int dop = fragment.getPipelineDop();
+                    if (dataSink instanceof OlapTableSink) {
+                        curTableSinkDop = dop;
+                    } else if (dataSink instanceof IcebergTableSink) {
```
Comment (ZiheLiu) [inline fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java:1749]: Is it possible that `forceSetTableSinkDop` is true but `dataSink` isn't `OlapTableSink` or `IcebergTableSink`?

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java:1749]: this variable affects only `pipeline_sink_dop` currently.  and the `pipeline_sink_dop` is only used for olap and iceberg table in fragment_executor.cpp.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 53 (67.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PlanFragment.java|2|4|50.00%|[269, 270]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|17|29|58.62%|[1693, 1695, 1696, 1698, 1699, 1700, 1702, 1705, 1747, 1748, 1749, 1751]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|6|8|75.00%|[218, 242]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|7|8|87.50%|[925]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|4|100.00%|[]|

**Review by zombee0 — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Pull request merged**